src/modules/csm/csm.py

```diff
@@ -119,16 +119,20 @@ def build_report(self, blockstamp: ReferenceBlockStamp) -> tuple:
                 shares[no_id] += amount
 
         tree = self.make_tree(shares)
-        cid: CID | None = None
+        tree_cid: CID | None = None
 
         if tree:
-            cid = self.w3.ipfs.publish(tree.encode())
+            state_cid = self.w3.ipfs.publish(self.state.encode())
+            logger.info({"msg": "State dump uploaded to IPFS", "cid": repr(state_cid)})
+            tree.state_cid = state_cid
```

Comment:
- Lets move publishing to ipfs to separate method
- instead of complicating Tree class let's just do smth like
```
state_cid = publish(state)
report_details_cid = publish({tree, state_cid})
```

Reply:
Changed to use "metadata".